### ISSUE TITLE
分布式单机多卡 fleet.distributed_optimizer(optimizer) 不同步

### ISSUE THREAD

**Issue by johnyanccer** [PaddlePaddle/Paddle#50589]
### bug描述 Describe the Bug
单机多卡训练，经过fleet.distributed_optimizer 封装的优化器仍然出现不同步，有的GPU的lr已经变小，有的仍然保持原样
paddle.device.set_device('gpu')
fleet.init(is_collective=True)

model = MyLayer()
model = fleet.distributed_model(model)
model.train()

# training settings
critron = ...
m = ...
lr_scheduler = paddle.optimizer.lr.ReduceOnPlateau(3.5e-4, factor=0.8, patience=                                                                                                                                   120, cooldown=18, min_lr=1e-7)   
decay_params = [
    p.name for n, p in model.named_parameters()
    if not any(nd in n for nd in ["bias", "norm"])
]
optim = paddle.optimizer.AdamW(learning_rate = lr_scheduler,
                               parameters = model.parameters(),
                               weight_decay=0.01,
                               apply_decay_param_fun=lambda x: x in decay_params                                                                                                                                   )
optim = fleet.distributed_optimizer(optim)

经过fleet.distributed_optimizer 封装的优化器仍然出现不同步，有的GPU的lr已经变小，有的仍然保持原样

### 其他补充信息 Additional Supplementary Information

paddlepaddle 2.4.1，单机多卡python -m paddle.distributed.launch --gpus=0,1 train.py

**Comment by cuicheng01** [comment]
目前是使用多机还是单机

**Comment by johnyanccer** [comment]
> 

分布式单机、多卡

**Comment by cuicheng01** [comment]
目前无法复现问题，可以给出更具体的代码和机器环境吗

**Comment by johnyanccer** [comment]
> 

已查明原因，目前paddle没有针对learning rate scheduler的同步机制，分布式训练中若采用ReduceOnPlateau | loss 自适应学习率衰减，则很可能不同worker调整lr的步伐不一致，最终导致模型难以收敛

解决办法：用paddle.distributed.all_reduce 显式同步各worker的loss值，由此同步调整所有worker的lr
